resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -194,6 +198,24 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
       sc.conf.getOption("spark.mesos.driver.frameworkId").map(_ + suffix)
     )
 
+    if (principal != null) {
+      logDebug(s"Principal found ($principal) starting token renewer")
+      val credentialRenewerThread = new Thread {
+        setName("MesosCredentialRenewer")
+        override def run(): Unit = {
+          val credentialRenewer =
+            new MesosCredentialRenewer(
+              conf,
+              hadoopDelegationTokenManager.get,
+              MesosCredentialRenewer.getTokenRenewalInterval(hadoopDelegationCreds.get, conf),
```

Comment:
hadoopDelegationCreds.get. Should we check against none? Creds are loaded in CoarseGrainedSchedulerBackend but if they are missing we should fail here?